pkg/rpc/server/server.go

```diff
@@ -224,55 +225,56 @@ func (s *Server) Start(errChan chan error) {
 		go func() {
 			ln, err := net.Listen("tcp", s.https.Addr)
 			if err != nil {
-				errChan <- err
+				s.errChan <- err
 				return
 			}
 			s.https.Addr = ln.Addr().String()
 			err = s.https.ServeTLS(ln, cfg.CertFile, cfg.KeyFile)
 			if err != http.ErrServerClosed {
 				s.log.Error("failed to start TLS RPC server", zap.Error(err))
-				errChan <- err
+				s.errChan <- err
 			}
 		}()
 	}
 	ln, err := net.Listen("tcp", s.Addr)
 	if err != nil {
-		errChan <- err
+		s.errChan <- err
 		return
 	}
 	s.Addr = ln.Addr().String() // set Addr to the actual address
 	go func() {
 		err = s.Serve(ln)
 		if err != http.ErrServerClosed {
 			s.log.Error("failed to start RPC server", zap.Error(err))
-			errChan <- err
+			s.errChan <- err
 		}
 	}()
 }
 
-// Shutdown overrides the http.Server Shutdown
-// method.
-func (s *Server) Shutdown() error {
+// Shutdown stops the RPC server. It can only be called once.
+func (s *Server) Shutdown() {
 	var httpsErr error
 
 	// Signal to websocket writer routines and handleSubEvents.
 	close(s.shutdown)
 
 	if s.config.TLSConfig.Enabled {
-		s.log.Info("shutting down rpc-server (https)", zap.String("endpoint", s.https.Addr))
+		s.log.Info("shutting down RPC server (https)", zap.String("endpoint", s.https.Addr))
 		httpsErr = s.https.Shutdown(context.Background())
 	}
 
-	s.log.Info("shutting down rpc-server", zap.String("endpoint", s.Addr))
+	s.log.Info("shutting down RPC server", zap.String("endpoint", s.Addr))
 	err := s.Server.Shutdown(context.Background())
 
 	// Wait for handleSubEvents to finish.
 	<-s.executionCh
 
-	if err == nil {
-		return httpsErr
+	if httpsErr != nil {
+		s.errChan <- httpsErr
```

Comment:
Shouldn't we return after that? Otherwice we need to be sure that the one who handles errors from `errChan` is able to receive two errors in a raw (both `httpsErr` and `err`) if so.

Reply:
Fixed.